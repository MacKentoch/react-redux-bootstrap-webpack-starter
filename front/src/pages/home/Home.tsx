import React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import Jumbotron from 'reactstrap/lib/Jumbotron';
import FadeInEntrance from '../../components/fadeInEntrance';
import HomeInfo from './styled/HomeInfo';
import MainTitle from './styled/MainTitle';
import { MappedDispatchToProps, MappedStateToProps, OwnProps } from './index';

// #region  types
export type Props = RouteComponentProps &
  MappedDispatchToProps &
  MappedStateToProps &
  OwnProps;
// #endregion

function Home(props: Props) {
  return (
    <FadeInEntrance>
      <Jumbotron>
        <HomeInfo>
          <MainTitle>ReactJS 16.13+ Bootstrap 4</MainTitle>
          <h2>
            with Hot Reload (<i>react-hot-loader 4+</i>
            )!!!
          </h2>
          <h2>and React Router v5</h2>
          <h2>and webpack 4.x</h2>
          <h1>Starter</h1>
          <p>
            <Link className="btn btn-success btn-lg" to={'/about'}>
              <i className="fa fa-info" />
              &nbsp; go to about
            </Link>
          </p>
        </HomeInfo>
      </Jumbotron>
    </FadeInEntrance>
  );
}

Home.displayName = 'Home';

export default Home;
