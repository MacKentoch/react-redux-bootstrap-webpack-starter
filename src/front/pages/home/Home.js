// @flow

// #region imports
import React, { PureComponent } from 'react';
import {
  type Match,
  type Location,
  type RouterHistory,
} from 'react-router-dom';
import { Link } from 'react-router-dom';
import Jumbotron from '../../components/jumbotron/Jumbotron';
import HomeInfo from './styled/HomeInfo';
import MainTitle from './styled/MainTitle';
import LightNote from './styled/LightNote';
// #endregion

// #region flow types
export type Props = {
  // react-router 4:
  match: Match,
  location: Location,
  history: RouterHistory,

  ...any,
};

export type State = any;
// #endregion

class Home extends PureComponent<Props, State> {
  // #region lifecycle
  render() {
    return (
      <div>
        <Jumbotron>
          <HomeInfo>
            <MainTitle>ReactJS 16.3+ Redux Bootstrap</MainTitle>
            <h2>
              with Hot Reload (<i>react-hot-loader 4+</i>)!!!
            </h2>
            <h2>and React Router v4</h2>
            <h2>and webpack 4.x</h2>
            <h2>
              and styled-components (<LightNote>
                so keep using SCSS like styles and benefit performant css-in-js
              </LightNote>)
            </h2>
            <h1>Starter</h1>
            <p>
              <Link className="btn btn-success btn-lg" to={'/about'}>
                <i className="fa fa-info" />
                &nbsp; go to about
              </Link>
            </p>
          </HomeInfo>
        </Jumbotron>
      </div>
    );
  }
  // #endregion
}

export default Home;
