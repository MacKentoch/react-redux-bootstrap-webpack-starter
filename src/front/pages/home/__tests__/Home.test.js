// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';
import Home from '../Home';
// #endregion

describe('Home page', () => {
  it('renders as expected', () => {
    const props = {
      actions: {},
    };
    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <Home {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
