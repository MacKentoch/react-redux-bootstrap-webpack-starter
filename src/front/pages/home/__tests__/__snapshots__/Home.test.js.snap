// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders as expected 1`] = `
<div>
  <div>
    <div
      className="jumbotron"
    >
      <div
        className="sc-bdVaJa bDWFJH"
      >
        <h1
          className="sc-bwzfXH jJLXBi"
        >
          ReactJS 16.3+ Redux Bootstrap
        </h1>
        <h2>
          with Hot Reload (
          <i>
            react-hot-loader 4+
          </i>
          )!!!
        </h2>
        <h2>
          and React Router v4
        </h2>
        <h2>
          and webpack 4.x
        </h2>
        <h2>
          and styled-components (
          <i
            className="sc-htpNat jfNlAg"
          >
            so keep using SCSS like styles and benefit performant css-in-js
          </i>
          )
        </h2>
        <h1>
          Starter
        </h1>
        <p>
          <a
            className="btn btn-success btn-lg"
            href="/about"
            onClick={[Function]}
          >
            <i
              className="fa fa-info"
            />
            Â  go to about
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
